src/types/magnitude.rs

```diff
@@ -71,7 +71,10 @@ impl Magnitude {
 
 impl From<BigUint> for Magnitude {
     fn from(value: BigUint) -> Self {
-        Magnitude::BigUInt(value)
+        match value.to_u64() {
```

Comment:
Minor, if my comment below is correct, I would put a small intent comment to make this more obvious.

```suggestion
        // prefer a compact representation for the magnitude
        match value.to_u64() {
```